lib/test.rb

```diff
@@ -725,40 +684,13 @@ def formula(formula_name)
       test "brew", "fetch", "--retry", *fetch_args
       test "brew", "uninstall", "--force", formula_name if formula.installed?
 
-      # shared_*_args are applied to both the main and --devel spec
-      shared_install_args = ["--verbose"]
-      shared_install_args << "--keep-tmp" if Homebrew.args.keep_tmp?
-      if !Homebrew.args.fast? &&
-         !Homebrew.args.no_bottle? &&
-         !formula.bottle_disabled?
-        shared_install_args << "--build-bottle"
-      end
-
-      # install_args is just for the main (stable, or devel if in a devel-only
-      # tap) spec
-      install_args = []
-      install_args << "--HEAD" if Homebrew.args.HEAD?
-
-      # Pass --devel or --HEAD to install in the event formulae lack stable.
-      # Supports devel-only/head-only.
-      # head-only should not have devel, but devel-only can have head.
-      # Stable can have all three.
-      formula_bottled = if devel_only_tap? formula
-        install_args << "--devel"
-        false
-      elsif head_only_tap? formula
-        install_args << "--HEAD"
-        false
-      else
-        formula.bottled?
-      end
-
-      install_args += shared_install_args
+      install_args = ["--verbose"]
+      install_args << "--build-bottle" unless formula.bottle_disabled?
       install_args << formula_name
 
       # Don't care about e.g. bottle failures for dependencies.
       install_passed = false
-      if !Homebrew.args.fast? || formula_bottled || formula.bottle_unneeded?
+      if formula.bottled? || formula.bottle_unneeded?
```

Comment:
This is a logic change. Previously this would always be true if `--fast` wasn't set. With this change, that is no longer the case.

This condition now evaluates to false for new formulae - they don't have a bottle block.